### PR TITLE
Use `dynalite` for DynamoDB simulation 

### PR DESCRIPTION
**Pull request by liushuyu**
This pull request actually contains multiple testing related changes:

- Use `dynalite` instead of `dynamodb-local` for the DynamoDB simulation (this will remove the Java requirements)
- Updated all the dependencies to the latest version to reduce security risks
- Added GitHub Actions CI to automate the testing process

### CODE REVIEW

**Review by James1218 — APPROVED** [review]
Looks good!

File: package.json [hunk 1]
```diff
@@ -6,37 +6,36 @@
...
     "prepublishOnly": "npm test && npm run lint",
     "preversion": "npm run lint",
     "version": "npm run format && git add -A src",
     "postversion": "git push && git push --tags",
-    "setupTestEnv": "cd test/local-dynamo && sls dynamodb install",
-    "startDynamo": "cd test/local-dynamo && sls dynamodb start",
-    "reseedTestEnv": "cd test/local-dynamo && sls dynamodb seed --seed=test"
+    "setupTestEnv": "true",
+    "startDynamo": "true",
+    "reseedTestEnv": "true"
```
Comment (ChenLi0830) [inline package.json:19]: Thanks for the MR! What does the "true" here do?

**Review by ChenLi0830 — APPROVED** [review]
Looks good! Approved with a minor question

File: package.json (hunk 1 above)
Comment (liushuyu) [reply package.json:19]: > Thanks for the MR! What does the "true" here do?

For compatibility reasons: `true` is a command in your system where it does nothing but exit with the status `0`.

If any external contributors have their own test setup, this change will ensure their setup remains working.

**Pull request merged**